app/views/shared/_assignments_dropdown_menu.html.erb

```diff
@@ -28,17 +28,19 @@
 
 <li class='section'><%= Assignment.model_name.human.pluralize %></li>
 
-<% assignments = @current_role.visible_assessments(assessment_type: 'Assignment') %>
-<% grade_entry_forms = @current_role.visible_assessments(assessment_type: 'GradeEntryForm') %>
+<% if @current_role%>
```

Comment:
I don't think this is the right place to make this check (a bit too far down the tree). Better to ensure the entire "sub_menu" doesn't render at all when `@current_role` is `nil`. I see in the `_sub_menu.html.erb` there's a policy check that should include this using `allowed_to?` so I'm not sure if this is an issue, but feel free to follow up with me about this on Slack with a more detailed error description. (But regardless you should revert the changes in this file.)